lib/matplotlib/backends/backend_gtk3.py

```diff
@@ -9,7 +9,12 @@ def fn_name(): return sys._getframe(1).f_code.co_name
 try:
     import gi
 except ImportError:
-    raise ImportError("Gtk3 backend requires pygobject to be installed.")
+    try:
+        import pgi
+        pgi.install_as_gi()
```

Comment:
ugh... let's not do it that way. I don't want to be in the position of
messing around with imports like that, because that can have unintended
side-effects for users who are unaware that matplotlib is mangling the
modules.

On Wed, Jul 29, 2015 at 9:07 AM, OceanWolf <notifications@github.com> wrote:

> In lib/matplotlib/backends/backend_gtk3.py
> <https://github.com/matplotlib/matplotlib/pull/4810#discussion_r35756016>:
>
> > @@ -9,7 +9,12 @@ def fn_name(): return sys._getframe(1).f_code.co_name
> >  try:
> >      import gi
> >  except ImportError:
> > -    raise ImportError("Gtk3 backend requires pygobject to be installed.")
> > +    try:
> > +        import pgi
> > +        pgi.install_as_gi()
>
> Well the code exists on github,
> https://github.com/lazka/pgi/blob/master/pgi/__init__.py
>
> it basically just does:
>
> import pgi
> sys.modules["gi"] = pgi
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/matplotlib/matplotlib/pull/4810/files#r35756016>.
>